### PR TITLE
Restructure multiline and grouped bar chart functions

### PR DESCRIPTION
**Pull request by MelinaGoula**
Functions to review:
- `horizontal_grouped_bar_chart `
- `vertical_grouped_bar_chart`
- `multi_linechart_test`
- helper functions (lines 1102 to 1490)

### CODE REVIEW

**Comment by MelinaGoula** [comment]
**Comparison of stacked bar chart functions** 

Previous (`stacked_chart`)
sphx_glr_plot_grouped_bar_rick_001
Rewrite (`horizontal_grouped_bar_chart`) 
sphx_glr_plot_grouped_bar_rick_002
Example of (`vertical_grouped_bar_chart`) 
sphx_glr_plot_grouped_bar_rick_003

**Comment by MelinaGoula** [comment]
Functionality that was not carried over from `multi_linechart` to `multi_linechart_test` (the temporary new version): 
- Ability to set custom styling parameters (font-size, font-family, fontfamily-list, stroke, stroke-width, border)
- Setting a plot title
- Dashed grid lines as opposed to solid grid lines 
- Customizable tick length and width 
- Customizable sizes for axis and tick labels 
- Customizable precision for tick labels 
- Ability to set a specific line colour, width and border style 
- Option to add shaded areas 

Example plot with `multi_linechart` showing a lot of the functionality 
sphx_glr_plot_multiline_003

Recreated with `multi_linechart_test`: 
sphx_glr_plot_multiline_002
I am not sure how to deal with legend placement in cases like this where there is not a lot of free space. One alternative is to follow the previous format of having multiple columns in the legend which would look like this: 
sphx_glr_plot_multiline_002

File: rick.py [hunk 1]
```diff
@@ -598,3 +855,901 @@ def bar_chart(data_in, xlab,**kwargs):
...
+            ax.yaxis.set_major_formatter(
+                mpl.ticker.StrMethodFormatter('{x:,' + precision + '}')
+            )
+        else:
+            ymin=0
+            ymax=np.max(ymax_array)*(1 + 0.1)
+    
+        delta = (ymax - ymin)/4
+        i = 0
+        while True:
```
Comment (gabrielwol) [inline rick.py:1004]: Should this be using `calculate_delta`? Same comment on line 580.

File: rick.py [hunk 2]
```diff
@@ -598,3 +855,901 @@ def bar_chart(data_in, xlab,**kwargs):
...
+                             else sett['body']['stroke-width']),
+                'border':('solid' if 'border' not in sett['body']
+                             else sett['body']['border'])
+            }
+        else:
+            dflt={
+                'font-size':12, 'font-family':'sans-serif',
+                'fontfamily-list':['Libre Franklin', 'DejaVu Sans'],
+                'stroke':'#000000', 'stroke-width':2, 'border':'solid'
+            }
```
Comment (gabrielwol) [inline rick.py:898]: Should default stylings live outside of a single graph function?

**Comment by gabrielwol** [comment]
> I am not sure how to deal with legend placement in cases like this where there is not a lot of free space. One alternative is to follow the previous format of having multiple columns in the legend which would look like this:

Have you considered having the legend always **outside** the plot? This could help with consistency no matter the data being plotted!

**Comment by gabrielwol** [comment]
> Functionality that was not carried over from multi_linechart to multi_linechart_test (the temporary new version):
....
Recreated with multi_linechart_test:
302367434-a85466f0-fda7-46b1-983d-2c67ca25f31e

I think the ticks would be an important feature for this graph type which we often use for timeseries. In our demos people are always trying to pinpoint the specific date that graphed events happen. Maybe dark ticks for the labelled points and lighter ticks for the others as a default?

**Comment by gabrielwol** [comment]
@MelinaGoula your 3 new charting functions are very elegant compared with the old versions, well done! See a few comments above.

File: rick.py [hunk 3]
```diff
@@ -598,3 +855,901 @@ def bar_chart(data_in, xlab,**kwargs):
...
+    if kwargs.get(f'{param_axis}max',None)==None:
+        upper=max_value 
+    else: 
+        upper=int(min_value+4*inc)
+    return max_value, min_value, inc, upper 
+
+def calculate_delta(max_value:float, min_value:float) -> (float, float):
+    '''
+    Returns parameters used to find the size of the y or x axis increments.
+    '''
```
[the remark this answers is not in the document]
Comment (MelinaGoula) [reply rick.py:1296]: Added!

**Comment by MelinaGoula** [comment]
I have made the following changes and I think this is ready for re-review:
- added the option to add minor ticks to the multi line chart function and specify their spacing, e.g.
image
- moved the legend to the outside of plots 
image
- added the option for labeled shaded areas, e.g. 
image

One thing I think can be done in the future to improve the code structure is to not have as many function parameters passed in the helper functions, but this can be probably only be done by refactoring the code.

**Review by gabrielwol — APPROVED** [review]
These changes look great! Well done @MelinaGoula

**Review by radumas — CHANGES_REQUESTED** [review]
Got a slight misfire in building Sphinx. 

We need the `docs/.nojekyll` file to remain.

[Also looks like we don't need to commit `.doctree` files (can add this to .gitignore) ](https://stackoverflow.com/a/33960721/4047679)

The contents of the `html` folder should be in `docs`.

Maybe we should change the build config back to building to `sphinx/build` and have instructions for people to manually copy the contents of `html` over to `docs` when they are ready to merge

**Review by radumas — APPROVED** [review]
Lots of changes here but I think this is really going to help clean up the repo!

**Pull request merged**